airflow/providers/cncf/kubernetes/executors/kubernetes_executor_utils.py

```diff
@@ -220,6 +220,16 @@ def process_status(
             # However, need to free the executor slot from the current executor.
             self.log.info("Event: pod %s adopted, annotations: %s", pod_name, annotations_string)
             self.watcher_queue.put((pod_name, namespace, ADOPTED, annotations, resource_version))
+        elif hasattr(pod.status, "reason") and pod.status.reason == "ProviderFailed":
```

Comment:
I take it the status is still pending here, so is the "ProviderFailed" reason permanent and will never actually run, or can it resolve itself?

Reply:
The reason I ask is we tolerate a lot of possibly temporary non-runnable pods already, e.g. if there are no nodes with space for the pod, and simply rely on the queued timeout to handle them. If it can be temporary, giving the pod a shot of getting going is probably the better behavior here.

Reply:
The pod provider failure is permanent. It will never succeeds.

Reply:
@jedcunningham ^^

Reply:
Thanks.